environs/manual/winrmprovisioner/winrmprovisioner.go

```diff
@@ -0,0 +1,581 @@
+// Copyright 2016 Canonical Ltd.
+// Copyright 2016 Cloudbase Solutions SRL
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package winrmprovisioner
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"html/template"
+	"io"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/juju/errors"
+	"github.com/juju/utils"
+	"github.com/juju/utils/arch"
+	"github.com/juju/utils/series"
+	"github.com/juju/utils/shell"
+	"github.com/juju/utils/winrm"
+
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/cloudconfig"
+	"github.com/juju/juju/cloudconfig/cloudinit"
+	"github.com/juju/juju/cloudconfig/instancecfg"
+	"github.com/juju/juju/environs/manual"
+	"github.com/juju/juju/instance"
+	"github.com/juju/juju/state/multiwatcher"
+)
+
+// detectJujudProcess powershell script to determine
+// if the jujud service is up and running in the machine
+// if it's up the script will output "Yes" if it's down
+// it will output "No"
+const detectJujudProcess = `
+	$jujuSvcs = Get-Service jujud-machine-*
+	if($jujuSvcs -and $jujuSvcs[0].Status -eq "running"){
+		return "Yes"
+	}
+	return "No"
+`
+
+// detectHardware is a powershell script that determines the following:
+//  - the processor architecture
```

Comment:
Why this is wierd? I think this comes to personal preference. I didn't come with a better way of explaining this powershell script in another manner. Also in my oppinion it's a good comment with main bullet points and very structured(all points explained). 
If have a different opinion please elaborate more.